### PR TITLE
Break prometheus-postgres-exporter into it's own directory

### PR DESCRIPTION
**Pull request by found-it**
### Image Size

- [x] The Image is smaller in size than its common public counterpart.
- [ ] The Image is larger in size than its common public counterpart (please explain in the notes).

Notes:

### Image Vulnerabilities

- [x] The Grype vulnerability scan returned 0 CVE(s).
- [ ] The Grype vulnerability scan returned > 0 CVE(s) (please explain in the notes).

Notes:

### Basic Testing - K8s cluster

- [x] The container image was successfully loaded into a kind cluster.
- [ ] The container image could not be loaded into a kind cluster (please explain in the notes).

Notes:

### Basic Testing - Package/Application

- [x] The application is accessible to the user/cluster/etc. after start-up.
- [ ] The application is not accessible to the user/cluster/etc. after start-up. (please explain in the notes).

Notes:

### Helm

- [x] A Helm chart has been provided and the container image can be used with the chart.  If needed, please add a -compat package to close any gaps with the public helm chart.
- [ ] A Helm chart has been provided and the container image is not working with the chart (please explain in the notes).
- [ ] A Helm chart was not provided.

Notes:

### Processor Architectures

- [x] The image was built and tested for x86_64.
- [ ] The image could not be built for x86_64 (please explain in the notes).
- [ ] The image was built and tested for aarch64.
- [ ] The image could not be built for aarch64. (please explain in the notes).

Notes:

### Functional Testing + Documentation

- [x] Functional tests have been included and the tests are passing.  All tests have been documnted in the notes section.

Notes:

### Environment Testing + Documentation

- [x] There has not been a request and/or there is no indication that this image needs tested on a public cloud provider.
- [ ] The container image has been tested successfully on a public cloud provider (AWS, GCP, Azure).
- [ ] The container image has not been tested successfully on a public cloud provider (AWS, GCP, Azure) (please explain in the notes).

Notes:

### Version

- [x] The package version is the latest version of the package.  The latest tag points to this version.
- [ ] The package version is the not the latest version of the package (please explain in the notes).

Notes:

### Dev Tag Availability

- [x] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [ ] There is not a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base') (please explain in the notes).

Notes:

### Access Control + Authentication

- [x] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
- [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [ ] The image requires a non-standard username or non-standard GID/UID (please explain in the notes).

### ENTRYPOINT

- [x] applications/servers/utilities set to call main program with no arguments e.g. [redis-server]
- [ ] applications/servers/utilities not set to call main program with no arguments e.g. [redis-server] (please explain in the notes)
- [ ] base images leave empty.
- [ ] base image and not empty (please explain in the notes).
- [ ] dev variants is set to entrypoint script that falls back to system.
- [ ] dev variants is not set to entrypoint script that falls back to system (please explain in the notes).

### CMD

- [x] For server applications give arguments to start in daemon mode (may be empty)
- [ ] For utilities/tooling bring up help e.g. `–help`
- [ ] For base images with a shell, call it e.g. [/bin/sh]

### Environment Variables

- [ ] Environment variables added.
- [x] Environment variables not added and not required.

### SIGTERM

- [x] The image responds to SIGTERM (e.g., `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`)

### Logs

- [x] Error logs write to stderr and normal logs to stdout. Logs DO NOT write to file.

### Documentation - README

- [x] A README file has been provided and it follows the README template.

### CODE REVIEW

File: images/prometheus-postgres-exporter/config/main.tf [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  value = jsonencode({
+    contents = {
+      packages = concat(["busybox"], var.extra_packages)
+    }
+    accounts = module.accts.block
+    entrypoint = {
+      command = "/usr/bin/postgres_exporter"
+    }
+  })
+}
```
[the remark this answers is not in the document]
Comment (found-it) [reply images/prometheus-postgres-exporter/config/main.tf:18]: Yeah standardizes our accounts and reduces the yaml!

File: images/prometheus-postgres-exporter/main.tf [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+module "latest-config" { source = "./config" }
+
+module "latest" {
+  source            = "../../tflib/publisher"
+  name              = basename(path.module)
+  target_repository = var.target_repository
+  config            = module.latest-config.config
+  build-dev         = true
+
```
[the remark this answers is not in the document]
Comment (found-it) [reply images/prometheus-postgres-exporter/main.tf:19]: Good catch, fixed

File: images/prometheus-postgres-exporter/tests/main.tf [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  namespace        = "postgres-exporter-bitnami-${random_pet.suffix.id}"
+  create_namespace = true
+
+  values = [jsonencode({
+    auth = {
+      postgresPassword = "${random_pet.suffix.id}"
+      username         = "${random_pet.suffix.id}"
+      password         = "${random_pet.suffix.id}"
+      database         = "${random_pet.suffix.id}"
+    }
```
Comment (cmdpdx) [inline images/prometheus-postgres-exporter/tests/main.tf:33]: Are these required to get the helm test to pass? I was able to pass it (with the older version of `postgres-exporter` so, grain of salt here) with just

```
values = [jsonencode({
    metrics = {
      image = {
        registry   = data.oci_string.ref.registry
        repository = data.oci_string.ref.repo
        tag        = data.oci_string.ref.pseudo_tag
      }
    }
  })]
```

Comment (found-it) [reply images/prometheus-postgres-exporter/tests/main.tf:33]: Yeah, the `metrics` container has to be `enabled` to spin up the postgres exporter. When I toggled that `enabled` flag it wanted me to specify a database and some creds so the exporter knows where to listen

Comment (found-it) [reply images/prometheus-postgres-exporter/tests/main.tf:33]: https://github.com/chainguard-images/images/pull/1647/files#diff-6c21e78c6356024de6b7ecfade2f1dc00fe68eb54c1ba4b3a2a271346fb748e1R35

**Review by cmdpdx — APPROVED** [review]

**Review by cmdpdx — APPROVED** [review]

**Review by pdeslaur — APPROVED** [review]

**Pull request merged**